test/misc/cancel-requests.js

```diff
@@ -0,0 +1,107 @@
+"use strict";
+const env = require("../..").env;
+const http = require("http");
+
+var routes = {
+  "/html": "<!DOCTYPE html><html>" +
+    "<head><script>window.test = true;</script><script src=\"/js\"></script></head>" +
+    "<body></body>" +
+    "</html>",
+  "/js": "window.testJs = true;",
+  "/xhr": "test"
+};
+
+exports["aborting env request should stop window creation"] = function (t) {
+
+  http.createServer(function (req, res) {
+    setTimeout(function () {
+      res.writeHead(200, { "Content-Length": routes[req.url].length });
+      res.end(routes[req.url]);
+    }, 200);
+  })
+  .listen(33333, function () {
+    var win = null;
+    var req = env({
+      url: "http://127.0.0.1:" + 33333 + "/html",
+      created: function (err, window) {
+        win = window;
+      },
+      features: {
+        FetchExternalResources: ["script"],
+        ProcessExternalResources: ["script"]
+      }
+    });
+    req.abort();
+    setTimeout(function () {
+      t.ok(!win, "the window should not have been created");
+      t.done();
+    }, 1000);
+  });
+
+};
+
+exports["closing window should close requests"] = function (t) {
+  http.createServer(function (req, res) {
+    setTimeout(function () {
+      res.writeHead(200, { "Content-Length": routes[req.url].length });
+      res.end(routes[req.url]);
+    }, 200);
+  })
+  .listen(33334, function () {
+    env({
+      url: "http://127.0.0.1:" + 33334 + "/html",
+      created: function (err, window) {
+        process.nextTick(function () {
+
+          var requestManager = window.document._requestManager;
```

Comment:
Tests should not test internal APIs. You need to find a different way of testing this.